h2o-algos/src/test/java/hex/tree/gbm/GBMInteractionConstraintsTest.java

```diff
@@ -0,0 +1,266 @@
+package hex.tree.gbm;
+
+import hex.Model;
+import hex.genmodel.CategoricalEncoding;
+import hex.genmodel.algos.tree.SharedTreeNode;
+import hex.tree.GlobalInteractionConstraints;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import water.DKV;
+import water.Scope;
+import water.TestUtil;
+import water.exceptions.H2OModelBuilderIllegalArgumentException;
+import water.fvec.Frame;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+import water.util.ArrayUtils;
+
+import java.util.*;
+
+
+@CloudSize(1)
+@RunWith(H2ORunner.class)
+public class GBMInteractionConstraintsTest extends TestUtil {
+    
+    @Test
+    public void testInteraction() {
+        String[][] interactions = new String[2][];
+        interactions[0] = new String[]{"AGE", "RACE", "PSA", "GLEASON"};
+        interactions[1] = new String[]{"DPROS", "DCAPS", "VOL"};
+        testInteractionConstraintsSimple(interactions);
+    }
+
+    @Test
+    public void testInteraction1() {
+        String[][] interactions = new String[2][];
+        interactions[0] = new String[]{"AGE", "RACE", "PSA", "GLEASON"};
+        interactions[1] = new String[]{"DPROS", "DCAPS", "VOL", "GLEASON"};
+        testInteractionConstraintsSimple(interactions);
+    }
+
+    @Test
+    public void testInteraction2() {
```

Comment:
Can we choose better name for these tests, to be able to see what is difference between testInteraction1 and testInteraction2? In this case DPROS can only interract with itself?

Reply:
These tests just test various combinations of features... It will name something like "testInteractionTwoSeparateInteraction" or "testInteractionTwoInteractionsOneFeatureInBothInteractions" etc... Maybe a comment would be better? I think it is readable from the parameter itself... :)

Reply:
Name of the test is good enough for me 👍